metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java

```diff
@@ -0,0 +1,25 @@
+package no.unit.nva.metadata;
+
+import org.apache.any23.validator.*;
+import org.w3c.dom.Node;
+
+import java.util.List;
```

Comment:
I guess I would understand this more intuitively when reading the code (if I understood your code correctly). I can't verify the functionality as there is no test.

```
package no.unit.nva.metadata;

import org.apache.any23.validator.DOMDocument;
import org.apache.any23.validator.Fix;
import org.apache.any23.validator.Rule;
import org.apache.any23.validator.RuleContext;
import org.w3c.dom.Node;



public class NoLowercaseDCPrefixInMetadataFix extends NoLowercase implements Fix {
    public static final String UPPERCASE_DC = "DC.";
    public static final String NO_LOWERCASE_DC_PREFIX_FIX = "no-lowercase-dc-prefix-in-metadata-fix";
    public static final int PREFIX_END = 3;

    @Override
    public String getHRName() {
        return NO_LOWERCASE_DC_PREFIX_FIX;
    }

    @Override
    public void execute(Rule rule, RuleContext context, DOMDocument document) {
        document.getNodes(HTML_META_TAGS_XPATH).forEach(this::fix);
    }

    private void fix(Node node) {
        Node nameNode = getMetaTagNameAttributeValue(node);
        if (containsLowerCaseDcPrefix(nameNode)) {
            nameNode.setTextContent(UPPERCASE_DC + nameNode.getTextContent().substring(PREFIX_END));
        }
    }
}
```

Reply:
And also

```
package no.unit.nva.metadata;

import org.w3c.dom.Node;

import static java.util.Objects.nonNull;

public class NoLowercase {
    public static final String HTML_META_TAGS_XPATH = "/HTML/HEAD/META";
    public static final String META_TAG_NAME_ATTRIBUTE = "name";
    public static final String LOWERCASE_DC = "dc.";

    protected Node getMetaTagNameAttributeValue(Node node) {
        return node.getAttributes().getNamedItem(META_TAG_NAME_ATTRIBUTE);
    }

    protected boolean containsLowerCaseDcPrefix(Node nameNode) {
        return nonNull(nameNode) && nameNode.getTextContent().startsWith(LOWERCASE_DC);
    }
}
```

Reply:
Hmm. I would not. I don't intuitively understand the difference between "execute" and "fix". Having both is confusing. Also it breaks with Apaches example (and arguably their interface) of how these classes are meant to be implemented.

Reply:
the private method is better for readability, but the name could be improved and is specific to the implementation.

Reply:
PR on my PR…meta :D